client/actions/assignment.js

```diff
@@ -0,0 +1,200 @@
+import * as selectors from '../selectors'
+
+/**
+ * Action dispatcher to load the list of assignments for current list settings.
+ * @param {string} filterBy - the filter by desk or user ('All', 'User')
+ * @param {string} searchQuery - the text used for free text query
+ * @param {string} orderByField - the field used to order the assignments ('Created', 'Updated')
+ * @param {string} orderDirection - the direction of order ('Asc', 'Desc')
+ */
+const loadAssignments = (filterBy, searchQuery, orderByField, orderDirection) =>
+    (dispatch) => {
+        dispatch({
+            type: 'CHANGE_LIST_SETTINGS',
+            payload: {
+                filterBy: filterBy,
+                searchQuery: searchQuery,
+                orderByField: orderByField,
+                orderDirection: orderDirection,
+                lastAssignmentLoadedPage: 1,
+            },
+        })
+
+        return dispatch(query({
+            filterBy: filterBy,
+            searchQuery: searchQuery,
+            orderByField: orderByField,
+            orderDirection: orderDirection,
+        }))
+        .then((data) => {
+            dispatch(receivedAssignments(data._items))
+            return Promise.resolve(data._items)
+        }, (error) => Promise.reject(error))
+    }
+
+/**
+ * Action dispatcher to load the next page of assignments.
+ * @param getState
+ */
+const loadMoreAssignments = () =>
+    (dispatch, getState) => {
+        const state = getState()
+        const {
+            filterBy,
+            searchQuery,
+            orderByField,
+            orderDirection,
+            lastAssignmentLoadedPage,
+        } =  selectors.getAssignmentListSettings(state)
+
+        dispatch({
+            type: 'CHANGE_LIST_SETTINGS',
+            payload: { lastAssignmentLoadedPage: lastAssignmentLoadedPage + 1 },
+        })
+
+        dispatch(query({
+            filterBy: filterBy,
+            searchQuery: searchQuery,
+            orderByField: orderByField,
+            orderDirection: orderDirection,
+            page: lastAssignmentLoadedPage + 1,
```

Comment:
imo this should use the updated state, what about first dispatching that `CHANGE_LIST_SETTINGS` and after that get the query params from state? actually query could get those params from state